implementation/src/test/java/io/smallrye/jwt/auth/principal/KeyLocationResolverTest.java

```diff
@@ -70,7 +77,28 @@ public void testLoadPemKeyWithWrongLocation() throws Exception {
     }
 
     @Test
-    public void testLoadHttpsJwks() throws Exception {
+    public void testLoadHttpsJwksMathchingKid() throws Exception {
+        JWTAuthContextInfo contextInfo = new JWTAuthContextInfo("https://github.com/my_key.jwks", "issuer");
+        contextInfo.setJwksRefreshInterval(10);
+
+        KeyLocationResolver keyLocationResolver = new KeyLocationResolver(contextInfo) {
+            protected HttpsJwks initializeHttpsJwks() {
+                return httpsJwks;
+            }
+        };
+        RsaJsonWebKey jwk = new RsaJsonWebKey(key);
+        jwk.setKeyId("1");
+        when(httpsJwks.getJsonWebKeys()).thenReturn(Collections.singletonList(jwk));
+        lenient().doThrow(new RuntimeException()).when(httpsJwks).refresh();
+        keyLocationResolver = Mockito.spy(keyLocationResolver);
+        when(signature.getHeaders()).thenReturn(headers);
+        when(headers.getStringHeaderValue(JsonWebKey.KEY_ID_PARAMETER)).thenReturn("1");
+        assertEquals(key, keyLocationResolver.resolveKey(signature, emptyList()));
+        assertNull(keyLocationResolver.verificationKey);
+    }
+
+    @Test
+    public void testLoadHttpsJwksNonMathchingKid() throws Exception {
```

Comment:
I'm a bit confused with this test. I don't think it is hitting the new code with the synchronization and call to `refresh`.

Reply:
@MikeEdgar do you refer to  `lenient()` ? Without it, if I just have `doNothing()` Mockito complains.
In `testLoadHttpsJwksMathchingKid`, `RuntimeException` would be thrown if `refresh` is called.

My be here I need to get `RuntimeException` thrown if `refresh` is not called, no idea right now if it is possible but will check. thanks

Reply:
@sberyozkin, the parts that are confusing me are the multiple calls to `setKeyId` with "1" and "2". I'm not clear that the call with the "2" has any effect. When I check the lines covered as reported by JaCoCo, it looks like the code to refresh to the token is not actually executed as I would expect based on the test method's name.

Reply:
@MikeEdgar I think I've fixed it and also added 1 more test to verify the refresh is declined if an attempt to made it too fast is done

Reply:
@sberyozkin - looks good. I see the new process is running now.